#!/usr/bin/env python
PACKAGE = "hector_motion_detection"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("motion_detect_threshold", int_t, 0, "Threshold ", 35, 0, 255)
gen.add("motion_detect_min_percept_size", double_t, 0, "Min Percept Size", 1, 0, 1000)
gen.add("motion_detect_max_percept_size", double_t, 0, "Max Percept Size", 300, 0, 1000)
gen.add("motion_detect_min_density", double_t, 0, "Density", 1, 0, 1000)
gen.add("percept_class_id", str_t, 0, "The percept class string", "victim")
gen.add("motion_detect_detectionLimit", int_t, 0, "Maximum Number of Motions to detect", 3, 0, 4);
gen.add("motion_detect_min_area", int_t, 0, "Minimal area of detected motions", 500, 0, 10000);
gen.add("motion_detect_max_area", int_t, 0, "Maximal area of detected motions", 100000, 0, 100000);
gen.add("motion_detect_erosion", int_t, 0, "Iterations for erosion on fgimg", 1, 0, 8);
gen.add("motion_detect_dilation", int_t, 0, "Iterations for dilation on fgimg", 1, 0, 8);

exit(gen.generate(PACKAGE, "motion_detection", "MotionDetection"))
